### PR TITLE
test cases addition mapsutil

### LINKED ISSUE #4: Implement tests for helper functions - mapsutil
## Description
Implement tests for the following functions:
- [x] Flatten
- [x] Walk

Check if these functions are aligned with nuclei implementation, otherwise update them and implement tests:
- [ ] HTTPToMap
- [x] DNSToMap
- [ ] HTTPRequesToMap
- [ ] HTTPResponseToMap

### PR DESCRIPTION
**Pull request by LuitelSamikshya**
closes #4

### CODE REVIEW

File: maps/mapsutil_test.go [hunk 1]
```diff
@@ -227,3 +237,56 @@ func TestClear(t *testing.T) {
 		require.Empty(t, m2)
 	})
 }
+
+func TestFlatten(t *testing.T) {
+	t.Run("Flatten (flat-map)", func(t *testing.T) {
+		input := make(map[string]interface{})
```
Comment (Mzack9999) [inline maps/mapsutil_test.go:243]: I think we can simplify with:
```go
input := map[string]any{"item": 0, "item1": 1, "item2": 2}
expected := maps.Clone(input)
```

File: maps/mapsutil_test.go [hunk 2]
```diff
@@ -227,3 +237,56 @@ func TestClear(t *testing.T) {
...
+		expected := map[string]any{"item": 0, "item1": 1, "item2": 2}
+		for i, v := range expected {
+			input[i] = v
+		}
+		result := Flatten(input, ".")
+		require.EqualValues(t, expected, result)
```
Comment (Mzack9999) [inline maps/mapsutil_test.go:249]: Maps iteration order is random, consider using `require.ElementsMatch` (ref: https://github.com/projectdiscovery/utils/issues/12) when comparing maps.

**Review by Mzack9999 — CHANGES_REQUESTED** [review]

File: maps/mapsutil_test.go (hunk 2 above)
Comment (LuitelSamikshya) [reply maps/mapsutil_test.go:247]: Used `require.ElementsMatch`

Comment (Mzack9999) [reply maps/mapsutil_test.go:247]: still using `require.EqualValues`

**Review by Mzack9999 — CHANGES_REQUESTED** [review]

**Review by Mzack9999 — APPROVED** [review]

**Pull request merged**